web/src/main/java/org/cbioportal/web/util/DataBinHelper.java

```diff
@@ -399,4 +405,47 @@ public Boolean areAllIntegers(Set<BigDecimal> uniqueValues) {
             .reduce((a, b) -> a && b)
             .orElse(false);
     }
+
+    public List<BigDecimal> generateBins(List<BigDecimal> sortedNumericalValues, BigDecimal binSize, BigDecimal anchorValue) {
+
+        Assert.notNull(sortedNumericalValues, "sortedNumerical values is null!");
+        Assert.notNull(binSize, "binSize values is null!");
+        Assert.notNull(anchorValue, "anchorValue values is null!");
+        
+        if (sortedNumericalValues.isEmpty()) {
+            return null;
+        }
+
+        // Assumes that elements are sorted in ascending order
+        BigDecimal minValue = sortedNumericalValues.get(0);
+        BigDecimal maxValue = sortedNumericalValues.get(sortedNumericalValues.size()-1);
+        Assert.isTrue(minValue.compareTo(maxValue) != 1, "minValue larger than maxValue. Input is not sorted in ascending order!");
```

Comment:
```suggestion
        Assert.isTrue(minValue.compareTo(maxValue) < 0, "minValue larger than maxValue. Input is not sorted in ascending order!");
```

That should make sonar happy